resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/registry/SegmentNode.java

```diff
@@ -414,11 +422,26 @@ public Match match(List<Match> matches, String httpMethod, HttpRequest request)
                else allowHeaderValue += ", ";
                allowHeaderValue += allow;
             }
-
+            
             if (httpMethod.equals("OPTIONS"))
             {
-               Response res = Response.ok(allowHeaderValue,  MediaType.TEXT_PLAIN_TYPE).header(HttpHeaderNames.ALLOW, allowHeaderValue).build();
-               throw new DefaultOptionsMethodException(Messages.MESSAGES.noResourceMethodFoundForOptions(), res);
+               ResponseBuilder resBuilder =  Response.ok(allowHeaderValue,  MediaType.TEXT_PLAIN_TYPE).header(HttpHeaderNames.ALLOW, allowHeaderValue);
+               
+               if (allowed.contains("PATCH"))
+               {  
+                  String acceptPatch = "";
+                  first = true;
+                  for (MediaType mediaType : patchAccepts)
+                  {
+                     if (first)
+                        first = false;
+                     else
+                        acceptPatch += ", ";
+                     acceptPatch += mediaType.toString();

```

Comment:
Please turn 'acceptPatch' into a StringBuilder and append on it instead of doing bunch of string concatenations (for performance reasons).

Reply:
Fixed both for Accept-patch and Allow headers.